microsoft/testsuites/cloud_hypervisor/ch_tests.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import Path
+from typing import Any
+
+from assertpy.assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    node_requirement,
+    schema,
+    search_space,
+)
+from lisa.tools import Lscpu
+from lisa.util import SkippedException
+from microsoft.testsuites.cloud_hypervisor.ch_tests_tool import CloudHypervisorTests
+
+
+@TestSuiteMetadata(
+    area="cloud-hypervisor",
+    category="community",
+    description="""
+    This test suite is for executing the tests maintained in the
+    upstream cloud-hypervisor repo.
+    """,
+)
+class CloudHypervisorTestSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node = kwargs["node"]
+        self._ensure_virtualization_enabled(node)
+
+    @TestCaseMetadata(
+        description="""
+            Runs cloud-hypervisor integration tests.
+        """,
+        priority=3,
+        requirement=node_requirement(
+            node=schema.NodeSpace(
+                core_count=16, memory_mb=search_space.IntRange(min=16 * 1024)
+            ),
+        ),
+    )
+    def verify_cloud_hypervisor_integration_tests(
+        self, log: Logger, node: Node, log_path: Path
+    ) -> None:
+        failures = node.tools[CloudHypervisorTests].run_tests("integration")
+        assert_that(failures).described_as("Unexpected failures").is_empty()
+
+    @TestCaseMetadata(
+        description="""
+            Runs cloud-hypervisor live migration tests.
+        """,
+        priority=3,
+        requirement=node_requirement(
+            node=schema.NodeSpace(
+                core_count=16, memory_mb=search_space.IntRange(min=16 * 1024)
+            ),
+        ),
+    )
+    def verify_cloud_hypervisor_live_migration_tests(
+        self, log: Logger, node: Node, log_path: Path
+    ) -> None:
+        failures = node.tools[CloudHypervisorTests].run_tests(
+            "integration-live-migration"
+        )
+        assert_that(failures).described_as("Unexpected failures").is_empty()
```

Comment:
we need using log to tell what happen, and what we can do in error message.

Reply:
Here also the default message feels sufficient to me especially after adding `described_as`. It prints "Unexpected failures" and also prints the list of tests that failed. I don't really see the need for any additional log.

Reply:
It's not possible to say in advance what caused a failure.  It can be due to many reasons which need to be determined by examining the logs.